src/modules/commander/PreflightCheck.cpp

```diff
@@ -682,6 +682,35 @@ static bool ekf2Check(orb_advert_t *mavlink_log_pub, vehicle_status_s &vehicle_s
 	return success;
 }
 
+static bool failureDetectorCheck(orb_advert_t *mavlink_log_pub, vehicle_status_s &status, bool report_fail, bool prearm)
+{
+	bool success = true;
+
+	if (!prearm) {
+		// Ignore failure detector check after arming.
+		return true;
+
+	} else {
+		if (status.failure_detector_status != 0) {
+			success = false;
+
+			if (report_fail) {
+				if (status.failure_detector_status & vehicle_status_s::FAILURE_ROLL) {
+					mavlink_log_critical(mavlink_log_pub, "PREFLIGHT FAIL: ROLL FAILURE DETECTED");
```

Comment:
```suggestion
					mavlink_log_critical(mavlink_log_pub, "Preflight Fail: Roll Failure detected");
```

Reply:
I was going to say the same thing. Don't shout please.